test/plugin_functional/plugins/core_plugin_deprecations/public/plugin.tsx

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { CoreSetup, Plugin, PluginInitializerContext } from 'kibana/public';
+
+declare global {
+  interface Window {
+    env?: PluginInitializerContext['env'];
+  }
+}
+
+export class CorePluginDeprecationsPlugin
```

Comment:
no need for client side app since i'm accessing the `core.deprecations` from the exposed `core` on the window object.

https://github.com/elastic/kibana/pull/94845/files#diff-b744ac6ce9320e70c9a0732e0b0464f7c9057f1fa71caee4142b0b9fb3a3cd51R91